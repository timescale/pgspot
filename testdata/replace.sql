-- test OR REPLACE for various create statements

CREATE OR REPLACE AGGREGATE unsafe_agg3(SFUNC=agg_sfunc,STYPE=internal);
CREATE OR REPLACE FUNCTION unsafe_func4() RETURNS TEXT LANGUAGE SQL AS $$ SELECT '123'; $$;
CREATE OR REPLACE PROCEDURE unsafe_proc5() LANGUAGE SQL AS $$ SELECT '123'; $$;
CREATE OR REPLACE TRANSFORM FOR type6 LANGUAGE lang6 (FROM SQL WITH FUNCTION hstore_to_plpython(internal), TO SQL WITH FUNCTION plpython_to_hstore(internal));
-- this is pg14 only which is not yet supported by sql parser
-- CREATE OR REPLACE TRIGGER trigger8 BEFORE INSERT ON table8 FOR EACH ROW EXECUTE PROCEDURE proc8();
CREATE OR REPLACE VIEW view9 AS SELECT pg_catalog.now();

