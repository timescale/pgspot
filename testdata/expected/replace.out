PS017: Unqualified object reference: unsafe_agg3
PS007: Unsafe object creation: unsafe_agg3()
PS002: Unsafe function creation: unsafe_func4()
PS005: Function without explicit search_path: unsafe_func4()
PS002: Unsafe function creation: unsafe_proc5()
PS005: Function without explicit search_path: unsafe_proc5()
PS006: Unsafe transform creation: type6
PS015: Unsafe view creation: view9
PS017: Unqualified object reference: view9

Errors: 4 Warnings: 5 Unknown: 0
